src/modules/universal-discovery/universal.discovery.module.js

```diff
@@ -236,19 +309,85 @@ export default class UniversalDiscoveryModule extends Component {
             onClick: this.handleConfirm.bind(this)
         };
 
+        if (this.state.activeTab === TAB_CREATE) {
+            return null;
+        }
+
         if (!this.state.selectedContent.length) {
             attrs.disabled = true;
         }
 
         return <button {...attrs}>{this.props.labels.udw.confirm}</button>
     }
 
+    checkPermission() {
+        if (this.state.isCheckingPermission === false) {
+            this.setState(state => Object.assign({}, state, {isCheckingPermission: true}));
+        }
+
+        checkCreatePermission({
+           token: this.props.restInfo.token,
+           contentTypeIdentifier: this.state.selectedContentType.identifier,
+           languageCode: this.state.selectedLanguage.languageCode,
+           locationId: this.state.contentMeta.id
+        }, (response) => {
+            this.setState(state => Object.assign({}, state, {isCheckingPermission: false}));
+
+            if (this.state.hasPermission !== response.access) {
+               this.setState(state => Object.assign({}, state, {hasPermission: response.access}));
```

Comment:
I think you should set state once, not twice in the same scope.

Reply:
I'm not sure how this could be in one. We want to change `isCheckingPermission` every time when the request is finished but change the permission state only when have to.

Reply:
What do you think about it?
```javascript
this.setState(state => Object.assign({}, state, {
    isCheckingPermission: false,
    hasPermission: state.hasPermission !== response.access ? response.access : state.hasPermission
}));
```

Reply:
I can try but as far as I know thanks to `immutable` with `Object.assign` this will make rerender of react component and this is what I wanted to avoid in this situation.

Reply:
You're already doing it, so why doing it twice?

Reply:
React will compare boolean values (I assume they are boolean). If there's any change then it will cause re-rendering of any related piece of code.